src/main/kotlin/org/digma/intellij/plugin/ui/jcef/BaseMessageRouterHandler.kt

```diff
@@ -353,5 +384,30 @@ abstract class BaseMessageRouterHandler(protected val project: Project) : Common
             setCurrentEnvironmentById(project, envId)
         }
     }
+
+    private fun login (requestJsonNode: JsonNode): LoginResult? {
+        val payload = getPayloadFromRequest(requestJsonNode)
+        val settingsState: SettingsState = SettingsState.getInstance()
+        val result = payload?.let {
+            try {
+                val provider = (RestAnalyticsProvider(
```

Comment:
in my opinion very strange to create a rest client here only for the purpose of login. you can keep the rest client on AuthManager and just call login. see comment above